vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -784,14 +794,15 @@ def collect_metrics(self, instance):
             for mor_name, mor in batch.iteritems():
                 if mor['mor_type'] == 'vm':
                     vm_count += 1
-                if 'metrics' not in mor or not mor['metrics']:
-                    continue
 
                 query_spec = vim.PerformanceManager.QuerySpec()
                 query_spec.entity = mor["mor"]
                 query_spec.intervalId = mor["interval"]
-                query_spec.metricId = mor["metrics"]
                 query_spec.maxSample = 1
+                if mor['mor_type'] in REALTIME_RESOURCES:
+                    query_spec.metricId = self.metadata_cache.get_metric_ids(i_key)
+                else:
+                    query_spec.metricId = mor.get("metrics")
```

Comment:
According to line 787 in the original file it seems this might be `None`? In that case we should probably not append the `query_spec`